benchmark-compare.js

```diff
@@ -39,6 +45,47 @@ if (!choices.length) {
     ])
   })
 
+  console.log(table.toString())
+} else if (commander.percentage) {
+  let data = []
+  choices.forEach(file => {
+    let content = readFileSync(`${resultsPath}/${file}.json`)
+    data.push(JSON.parse(content.toString()))
+  })
+  data.sort((a, b) => {
+    return parseFloat(b.requests.mean) - parseFloat(a.requests.mean)
+  })
+  const base = Object.assign({}, {
+    name: data[0].server,
+    request: data[0].requests.mean,
+    latency: data[0].latency.mean,
+    throughput: data[0].throughput.mean
+  })
+  const table = new Table({
+    head: [
+      '',
+      'Version',
+      'Router',
+      `Requests/s\n(% of ${base.name})`,
+      `Latency\n(% of ${base.name})`,
+      `Throughput/Mb\n(% of ${base.name})`
+    ]
+  })
+  data.forEach(result => {
+    const beBold = result.server === 'fastify'
+    const { version = 'N/A', hasRouter = false } = info(result.server) || {}
+    const getPct = (base, value) => ((value / base * 100).toFixed(2))
+
+    table.push([
+      bold(beBold, chalk.blue(result.server)),
+      bold(beBold, version),
+      bold(beBold, hasRouter ? '✓' : '✗'),
+      bold(beBold, `${result.requests.mean}\n(${getPct(base.request, result.requests.mean)})`),
+      bold(beBold, `${result.latency.mean}\n(${getPct(base.latency, result.latency.mean)})`),
+      bold(beBold, `${(result.throughput.mean / 1024 / 1024).toFixed(2)}\n(${getPct(base.throughput, result.throughput.mean)})`)
```

Comment:
is there a way to reuse some of the code for the `-t` swtich, to avoid code duplication?

Reply:
As first step create a `createRow` helper function. But I am afraid it will damage more the not so good readability of the code.

Reply:
No worries, it's ok as it is.